src/views/css/main.ts

```diff
@@ -253,39 +243,80 @@ export const statusBar = {
     },
 };
 
-export const sessions = (sessionsCount: number) => Object.assign(
+export const sessions = () => Object.assign(
     {
         backgroundColor: backgroundColor,
     },
-    sessionsGrid.container(sessionsCount),
+    sessionsGrid.container(),
 );
 
-export const session = (isActive: boolean, index: number) => {
+export const session = (isActive: boolean, containerViewMapLeaf: ViewMapLeaf<Session>) => {
     const styles: CSSObject = {
         position: "relative",
         outline: "none",
+        flex: 1,
     };
 
+    if (containerViewMapLeaf !== undefined) {
+        if (containerViewMapLeaf.containerType === ContainerType.Column) {
+          styles.height = `calc(100% / ${containerViewMapLeaf.childs.length})`;
+        }
+    } else {
+      styles.height = `calc(${sessionsHeight})`;
+    }
+
     if (!isActive) {
         styles.boxShadow = `0 0 0 1px ${alpha(colors.white, 0.3)}`;
-        styles.margin = "0 0 1px 0px";
+        styles.margin = "0 1px 0 0";
     }
 
-    return Object.assign(
-        styles,
-        sessionsGrid.session(index),
-    );
+    return styles;
 };
 
-export const sessionShutter = (index: number) => Object.assign(
-    {
-        backgroundColor: colors.white,
-        zIndex: 1,
-        opacity: 0.2,
-        pointerEvents: "none",
-    },
-    sessionsGrid.sessionShutter(index),
-);
+export const sessionContainer = (viewMapLeaf: ViewMapLeaf<Session>, parentContainerViewMapLeaf: ViewMapLeaf<Session>) => {
+  const styles: CSSObject = {
+    display: "flex",
+    flex: 1,
+    overflowY: "scroll",
+  };
+
+  if (parentContainerViewMapLeaf !== undefined) {
+      if (parentContainerViewMapLeaf.containerType === ContainerType.Column) {
+        styles.height = `calc(${sessionsHeight} / ${parentContainerViewMapLeaf.childs.length})`;
+      } else {
+        styles.height = `calc(${sessionsHeight})`;
+      }
+  } else {
+    styles.height = `calc(${sessionsHeight})`;
+  }
+
+  if (viewMapLeaf.containerType === ContainerType.Column) {
+    styles.flexDirection = "column";
+    styles.height = "100%";
+  } else {
+    styles.flexDirection = "row";
+  }
+
+  return styles;
+};
+
+export const sessionShutter = {
+    backgroundColor: colors.white,
+    zIndex: 1,
+    opacity: 0.2,
+    pointerEvents: "none",
+    position: "absolute",
+    top: 0,
+    bottom: 0,
+    left: 0,
+    right: 0,
+};
+
+export const jobsContainer = {
+  height: "100%",
+  width: "100%",
+  overflowY: "scroll",
```

Comment:
put `shutter` inside `Session`, because it is related to `Session` actually. So `shutter` will gets all the space of `Session` to cover. But `jobsContainer` leaves still scrollable